### PR TITLE
fix(autoware_rtc_interface): fix constParameterReference

### PR DESCRIPTION
**Pull request by kobayu858**
## Description
This is a fix based on cppcheck constParameterReference warnings
```
planning/autoware_rtc_interface/src/rtc_interface.cpp:182:24: style: Parameter 's' can be declared as reference to const [constParameterReference]
      [command](auto & s) { return s.uuid == command.uuid; });
                       ^

planning/autoware_rtc_interface/src/rtc_interface.cpp:211:24: style: Parameter 's' can be declared as reference to const [constParameterReference]
      [command](auto & s) { return s.uuid == command.uuid; });
                       ^

planning/autoware_rtc_interface/src/rtc_interface.cpp:251:19: style: Parameter 's' can be declared as reference to const [constParameterReference]
    [uuid](auto & s) { return s.uuid == uuid; });
                  ^

planning/autoware_rtc_interface/src/rtc_interface.cpp:284:19: style: Parameter 's' can be declared as reference to const [constParameterReference]
    [uuid](auto & s) { return s.uuid == uuid; });
                  ^

planning/autoware_rtc_interface/src/rtc_interface.cpp:300:69: style: Parameter 's' can be declared as reference to const [constParameterReference]
    stored_commands_.begin(), stored_commands_.end(), [uuid](auto & s) { return s.uuid == uuid; });
                                                                    ^

planning/autoware_rtc_interface/src/rtc_interface.cpp:330:19: style: Parameter 's' can be declared as reference to const [constParameterReference]
    [uuid](auto & s) { return s.uuid == uuid; });
                  ^

planning/autoware_rtc_interface/src/rtc_interface.cpp:377:19: style: Parameter 's' can be declared as reference to const [constParameterReference]
    [uuid](auto & s) { return s.uuid == uuid; });
                  ^

planning/autoware_rtc_interface/src/rtc_interface.cpp:386:19: style: Parameter 's' can be declared as reference to const [constParameterReference]
    [uuid](auto & s) { return s.uuid == uuid; });
                  ^

planning/autoware_rtc_interface/src/rtc_interface.cpp:402:19: style: Parameter 's' can be declared as reference to const [constParameterReference]
    [uuid](auto & s) { return s.uuid == uuid; });
                  ^

```
## Related links

**Parent Issue:**

- Link

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/autoware_rtc_interface/src/rtc_interface.cpp [hunk 1]
```diff
@@ -151,7 +151,7 @@ std::vector<CooperateResponse> RTCInterface::validateCooperateCommands(
 
     const auto itr = std::find_if(
       registered_status_.statuses.begin(), registered_status_.statuses.end(),
-      [command](auto & s) { return s.uuid == command.uuid; });
+      [command](const auto & s) { return s.uuid == command.uuid; });
```
Comment (veqcc) [inline planning/autoware_rtc_interface/src/rtc_interface.cpp:154]: @kobayu858 
Are you 100% sure it is OK to make `s` a const reference even though `s` is modified?

**Review by veqcc — APPROVED** [review]

**Comment by kobayu858** [comment]
One new warning was added and addressed.

**Comment by kobayu858** [comment]
@rej55 @kyoichi-sugahara @satoshi-ota @taikitanaka3 

Please confirm the PR. Thank you very much.

**Review by rej55 — APPROVED** [review]
LGTM
Thank you!

**Comment by kobayu858** [comment]
Thank you for merging.

**Pull request merged**